console/services/market_app_service.py

```diff
@@ -67,6 +69,17 @@ def install_service(self,
         try:
             app_templates = json.loads(market_app_version.app_template)
             apps = app_templates["apps"]
+
+            # check resources
+            data = team_services.get_tenant_resource(tenant, region)
```

Comment:
There is a detection interface, send the required memory, and the result will be returned to you. No need to acquire all resources and then calculate